src/toxcore/dht/packet/dht_request.rs

```diff
@@ -264,9 +273,89 @@ impl ToBytes for DhtPkAnnounce {
     }
 }
 
+/** Hardening nodes request of DHT Request packet.
+
+Length    | Content
+--------- | -------------------------
+`1`       | `0x30`
+`1`       | `0x02`
+`51`      | `PackedNode`
+`32`      | Search Id (PublicKey)
+
+*/
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub struct HardeningRequest {
+    /// node to be checked
+    pub node: PackedNode,
+    /// Search id
+    pub id: PublicKey,
+}
+
+impl FromBytes for HardeningRequest {
+    named!(from_bytes<HardeningRequest>, do_parse!(
+        tag!("\x30") >>
+        tag!("\x02") >>
+        node: call!(PackedNode::from_bytes) >>
```

Comment:
It's not PackedNode format, they just use memcpy of plain c struct:
https://github.com/TokTok/c-toxcore/blob/744dc2f5da2c54ad1e4d7d6ce229bc1756d81263/toxcore/DHT.c#L2442
So I'd suggest not to parse it (or parse as a byte array). Anyway it seems it's going to be removed from `c-toxcore`.

Reply:
https://github.com/TokTok/c-toxcore/blob/744dc2f5da2c54ad1e4d7d6ce229bc1756d81263/toxcore/DHT.c#L2439

It is actually parsed.

Reply:
https://github.com/TokTok/c-toxcore/blob/744dc2f5da2c54ad1e4d7d6ce229bc1756d81263/toxcore/DHT.c#L2439

It is actually parsed.

Reply:
https://github.com/TokTok/c-toxcore/blob/744dc2f5da2c54ad1e4d7d6ce229bc1756d81263/toxcore/DHT.c#L2439

It is actually parsed.

Reply:
https://github.com/TokTok/c-toxcore/blob/744dc2f5da2c54ad1e4d7d6ce229bc1756d81263/toxcore/DHT.c#L2439

It is actually parsed.

Reply:
@NamsooCho unfortunately I removed all duplicate comments + the last one. Sorry)

Basically we don't care. All we need is id's and the length of the structs. We don't need to parse the underlying struct correctly, we need to create a packet for handling (fake fandling =) ))

Reply:
@NamsooCho unfortunately I removed all duplicate comments + the last one. Sorry)

Basically we don't care. All we need is id's and the length of the structs. We don't need to parse the underlying struct correctly, we need to create a packet for handling (fake handling =) ))

Reply:
Basically we don't care. All we need is id's and the length of the structs. We don't need to parse the underlying struct correctly, we need to create a packet for handling (fake handling =) ))

Reply:
Basically we don't care. All we need is id's and the length of the structs. We don't need to parse the underlying struct correctly, we need to create a packet for handling (fake handling =) ))

Reply:
Basically we don't care. All we need is id's and the length of the structs. We don't need to parse the underlying struct correctly, we need to create a packet for handling (fake handling =) ))

Reply:
Basically we don't care. All we need is id's and the length of the structs. We don't need to parse the underlying struct correctly, we need to create a packet for handling (fake handling =) ))

Reply:
done